tests/test_extract_child_mesh.cpp

```diff
@@ -0,0 +1,589 @@
+        }
+
+        std::shared_ptr<Mesh> child_ptr =
+            wmtk::multimesh::utils::extract_and_register_child_mesh_from_tag(
+                parent,
+                "is_child",
+                1,
+                PE);
+
+        const auto& p_mul_manager = parent.multi_mesh_manager();
+        REQUIRE(p_mul_manager.children().size() == 1);
+
+        const auto& child = *(p_mul_manager.children()[0].mesh);
+        REQUIRE(child.get_all(PE).size() == 3);
+        REQUIRE(child.get_all(PV).size() == 3);
+    }
+    SECTION("two_neighbors")
+    {
+        DEBUG_TriMesh parent = two_neighbors();
+        auto tag_handle = parent.register_attribute<long>("is_child", wmtk::PrimitiveType::Edge, 1);
+        auto tag_accessor = parent.create_accessor(tag_handle);
+
+        const Tuple& e01 = parent.edge_tuple_between_v1_v2(0, 1, 0);
+        const Tuple& e02 = parent.edge_tuple_between_v1_v2(0, 2, 0);
+        const Tuple& e12 = parent.edge_tuple_between_v1_v2(1, 2, 0);
+        const Tuple& e03 = parent.edge_tuple_between_v1_v2(0, 3, 1);
+        const Tuple& e13 = parent.edge_tuple_between_v1_v2(1, 3, 1);
+        const Tuple& e04 = parent.edge_tuple_between_v1_v2(0, 4, 2);
+        const Tuple& e24 = parent.edge_tuple_between_v1_v2(2, 4, 2);
+
+        tag_accessor.scalar_attribute(e01) = 1;
+        tag_accessor.scalar_attribute(e02) = 1;
+        tag_accessor.scalar_attribute(e12) = 1;
+        tag_accessor.scalar_attribute(e03) = 2;
+        tag_accessor.scalar_attribute(e13) = 2;
+        tag_accessor.scalar_attribute(e04) = 3;
+        tag_accessor.scalar_attribute(e24) = 3;
+
+
+        std::shared_ptr<Mesh> child_ptr0 =
+            wmtk::multimesh::utils::extract_and_register_child_mesh_from_tag(
+                parent,
+                "is_child",
+                1,
+                PE);
+        std::shared_ptr<Mesh> child_ptr1 =
+            wmtk::multimesh::utils::extract_and_register_child_mesh_from_tag(
+                parent,
+                "is_child",
+                2,
+                PE);
+        std::shared_ptr<Mesh> child_ptr2 =
+            wmtk::multimesh::utils::extract_and_register_child_mesh_from_tag(
+                parent,
+                "is_child",
+                3,
+                PE);
+
+
+        const auto& p_mul_manager = parent.multi_mesh_manager();
+        REQUIRE(p_mul_manager.children().size() == 3);
+
+        const auto& child0 = *(p_mul_manager.children()[0].mesh);
+        const auto& child1 = *(p_mul_manager.children()[1].mesh);
+        const auto& child2 = *(p_mul_manager.children()[2].mesh);
+        REQUIRE(child0.get_all(PE).size() == 3);
+        REQUIRE(child0.get_all(PV).size() == 3);
+        REQUIRE(child1.get_all(PE).size() == 2);
+        REQUIRE(child1.get_all(PV).size() == 3);
+        REQUIRE(child2.get_all(PE).size() == 2);
+        REQUIRE(child2.get_all(PV).size() == 3);
+    }
+}
+
+//     auto& child0 = *child0_ptr;
```

Comment:
What's with all that commented out code? Where does it come from and why is it here?